libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_JRE_Serial.h"
+
+#if AP_RANGEFINDER_JRE_SERIAL_ENABLED
+
+#include <GCS_MAVLink/GCS.h>
+
+#define FRAME_HEADER_1   0x52  // 'R'
+#define FRAME_HEADER_2   0x41  // 'A'
+
+#define DIST_MAX_CM 50000
+
+#define CCITT_INITPARA   0xffff
+#define CCITT_POLYNOMIAL 0x8408
+#define CCITT_OUTPARA    0xffff
+
+/* CRC16CCITT */
+static uint16_t CalcCRC16CCITT(uint8_t *cbuffer, uint16_t csize)
```

Comment:
There are already two in CHIBIOS.
I think it would be better to add them below and share them with you for future use.

libraries/AP_Math/crc.cpp

Reply:
I agree withe @muramura that **if possible** we should re-use one of the existing crc methods ([see here](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/crc.h#L22))

Reply:
I moved crc calculation to libraries/AP_Math/crc.cpp.